docs/user-guide/release-notes/2.11.x.rst

```diff
@@ -0,0 +1,13 @@
+=======================
+Pulp 2.11 Release Notes
+=======================
+
+Pulp 2.11.0
+===========
+
+New Features
+------------
+
+* Re-sync of the repository will be a no operation by default if since the last sync, no units
```

Comment:
@dralley re-sync of the repository cannot be no operational, therefore have skipped result based on your code and the feature requirements. The improvements that you did will do always an *operational* sync, the difference is - sometimes it will perform all sync steps without skipping any and sometimes it will take advantage of sync optimization logic and skip some of the steps. What you did is - you moved part of logic into the platform, so we first will look into platform at some conditions, if those conditions are true then we will disregard the rest of logic optimization that is done in each plugin. and do a so-called force sync.